src/Microsoft.DotNet.ProjectModel.Loader/ProjectLoadContext.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Microsoft.DotNet.ProjectModel.Loader
+{
+    /// <summary>
+    /// An <see cref="AssemblyLoadContext"/> designed to allow loading from a Project.
+    /// </summary>
+    public class ProjectLoadContext : AssemblyLoadContext
+    {
+        private readonly IDictionary<AssemblyName, string> _assemblyPaths;
+        private readonly IDictionary<string, string> _nativeLibraries;
+        private readonly IEnumerable<string> _searchPaths;
+
+        private static readonly string[] NativeLibraryExtensions;
+        private static readonly string[] ManagedAssemblyExtensions = new[]
```

Comment:
Can we factor this out to a separate class? this seems like reference data we'll want to use in other places.